### PR TITLE
Move files in sgrep_lint into sgrep_lint/semgrep

### PR DESCRIPTION
**Pull request by brendongo**
This namespaces all local sgrep-lint code in the semgrep module for absolute imports.

Also does some minor refactoring to break up some monolithic functions.

No change in functionality. Mostly just moving code around.

### CODE REVIEW

**Comment by ievans** [comment]
@brendongo if you do this, it becomes extremely annoying to test locally. Eg `python3 sgrep.py` no longer works. Will you provide clear directions on how to run locally to make that super easy for contributors?

**Comment by mschwager** [comment]
> @brendongo if you do this, it becomes extremely annoying to test locally. Eg `python3 sgrep.py` no longer works. Will you provide clear directions on how to run locally to make that super easy for contributors?

The cool thing about `__main__.py` is you can now easily run it as a Python module with `python -m`. E.g.

```
$ ls
foo
$ cat foo/__main__.py 
print("HELLO WORLD!")
$ python3 -m foo
HELLO WORLD!
```

File: sgrep_lint/semgrep/__main__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
+import sys
+
+from semgrep.cli import cli
```
Comment (DrewDennison) [inline sgrep_lint/semgrep/__main__.py:4]: nice

File: sgrep_lint/tests/run-lint-tests.sh [hunk 2]
```diff
@@ -101,7 +101,7 @@ cd "${THIS_DIR}"
 PYTHONPATH=.. pytest .
 
 local_tests() {
-    SGREP="./sgrep.py"
+    SGREP="./semgrep/__main__.py"
```
Comment (DrewDennison) [inline sgrep_lint/tests/run-lint-tests.sh]: is `__main__.py` standard practice? I was expecting to see `semgrep/semgrep.py` or something

**Review by DrewDennison — DISMISSED** [review]

**Comment by ievans** [comment]
awesome @mschwager , can that be documented somewhere?

File: sgrep_lint/tests/run-lint-tests.sh (hunk 2 above)
Comment (brendongo) [reply sgrep_lint/tests/run-lint-tests.sh]: Yeahp. If you want to run from source. __main__.py runs when you run the module. When we package this for people `semgrep` will work. See @mschwager comment above.

**Comment by brendongo** [comment]
> awesome @mschwager , can that be documented somewhere?

I'll add it to the development.md file? Seems like this shouldn't be in the main readme.

**Pull request merged**